src/DataWrangling/EN4/EN4_metadata.jl

```diff
@@ -0,0 +1,186 @@
+    shortname = short_name(metadata)
+    yearstr  = string(Dates.year(metadata.dates))
+    monthstr = string(Dates.month(metadata.dates), pad=2)
+    return "EN.4.2.2.f.analysis.g10." * yearstr * lpad(string(monthstr), 2, '0') * ".nc"
+end
+
+# Convenience functions
+short_name(data::Metadata{<:EN4Monthly}) = EN4_short_names[data.name]
+
+location(data::EN4Metadata) = EN4_location[data.name]
+
+dataset_temperature_units(data::Metadata{<:EN4Monthly}) = Kelvin()
+
+variable_is_three_dimensional(data::EN4Metadata) =
+    data.name == :temperature ||
+    data.name == :salinity
+
+EN4_short_names = Dict(
+    :temperature           => "temperature",
+    :salinity              => "salinity"
+)
+
+EN4_location = Dict(
+    :temperature           => (Center, Center, Center),
+    :salinity              => (Center, Center, Center)
+)
+
+function metadata_url(m::Metadata{<:EN4Monthly})
+    year = string(Dates.year(m.dates))
+    if Dates.year(m.dates) < 2021
+        return EN4_url_pre2021 * year * ".zip"
+    else
+        return EN4_url_post2021 * year * ".zip"
+    end
+end
+
+## This function is explicitly for the downloader to check if the zip file/extracted file exists,
+## then to download the relevant URL (from above)
+
+function metadata_path_EN4(m::Metadata{<:EN4Monthly})
+    year = string(Dates.year(m.dates))
+    month = string(Dates.month(m.dates))
+    zipfile = m.dir * "EN4_" * year * ".zip"
+    extracted_file = m.dir * "EN.4.2.2.f.analysis.g10." * year * lpad(string(month), 2, '0') * ".nc"
+    return zipfile, extracted_file
+end
+
+function unzip(file, exdir="")
+    fileFullPath = isabspath(file) ?  file : joinpath(pwd(),file)
+    basePath = dirname(fileFullPath)
+    outPath = (exdir == "" ? basePath : (isabspath(exdir) ? exdir : joinpath(pwd(),exdir)))
+    isdir(outPath) ? "" : mkdir(outPath)
+    zarchive = ZipFile.Reader(fileFullPath)
+    for f in zarchive.files
+        fullFilePath = joinpath(outPath,f.name)
+        if (endswith(f.name, "/") || endswith(f.name, "\\"))
+            mkdir(fullFilePath)
+        else
+            write(fullFilePath, read(f))
+        end
+    end
+    close(zarchive)
+end
+
+"""
+    download_dataset(metadata::EN4Metadata; url = urls(metadata))
+
+Download the dataset specified by the `metadata::EN4Metadata`. If `metadata.dates` is a single date,
+the dataset is downloaded directly. If `metadata.dates` is a vector of dates, each date
+is downloaded individually.
+
+The data download requires a username and password to be provided in the `EN4_USERNAME` and
+`EN4_PASSWORD` environment variables respectively. This can be done by exporting the
+environment variables in the shell before running the script, or by launching julia with
+
+```
+EN4_USERNAME=myusername EN4_PASSWORD=mypassword julia
+```
+
+or by invoking
+
+```julia
+julia> ENV["EN4_USERNAME"] = "myusername"
+
+julia> ENV["EN4_PASSWORD"] = "mypassword"
+```
+
+within julia.
```

Comment:
@taimoorsohail do we need this? I haven't set any EN4_USERNAME/EN4_PASSWORD and was able to download things.